app/graphql/mutations/create_user_standing.rb

```diff
@@ -0,0 +1,75 @@
+module Mutations
+  class CreateUserStanding < ApplicationQuery
+    include QueryAuthorizeSchoolAdmin
+    argument :student_id, ID, required: true
+    argument :reason, String, required: true
+    argument :standing_id, ID, required: true
+
+    description "Create a new standing log"
+
+    field :user_standing, Types::UserStandingType, null: true
+
+    def resolve(_params)
+      notify(
+        :success,
+        I18n.t("shared.notifications.done_exclamation"),
+        I18n.t("mutations.create_user_standing.success_notification")
+      )
+      { user_standing: create_user_standing }
+    end
+
+    private
+
+    def create_user_standing
+      UserStanding.transaction do
+        user_standing =
+          UserStanding.create!(
+            user_id: student.user.id,
+            reason: @params[:reason],
+            standing_id: @params[:standing_id],
+            creator: current_user
```

Comment:
The `transaction` shouldn't` care about sending email;

Reply:
> The transaction shouldn't` care about sending email;

Why?

But that is the pattern we are following across many places in the codebase.
`pupilfirst/app/queries/create_school_admin_mutator.rb`
```rb
  def create_school_admin
    SchoolAdmin.transaction do
      user =
        persisted_user ||
          current_school.users.create!(email: email, title: "School Admin")
      user.update!(name: name)
      new_school_admin = current_school.school_admins.create!(user: user)

      current_school
        .school_admins
        .where.not(user_id: user.id)
        .each do |admin|
          SchoolAdminMailer.school_admin_added(
            admin,
            new_school_admin,
            current_user
          ).deliver_later
        end

      create_audit_record(user)
      new_school_admin
    end
  end
  ```

---

`create_feedback.rb`

```rb
    def create_feedback
      StartupFeedback.transaction do
        startup_feedback =
          StartupFeedback.create!(
            feedback: @params[:feedback],
            faculty: coach,
            timeline_event: submission
          )
        StartupFeedbackModule::EmailService.new(startup_feedback).send
      end
    end
```

we are doing the same in the `create_grading` mutation and may be at other places as well ?

Reply:
It would mostly be old code; 

Transactions are ideal when you are doing multiple database writes, it ensures that all or none is saved;
In this specific case, why should we block the write waiting for a network request? (sending email).

Reply:
Thank you 

Fixed in this https://github.com/pupilfirst/pupilfirst/pull/1468/commits/1b441d7d71a47229a95b062e52502b0818d101fd